bin/picom-trans

```diff
@@ -46,194 +55,305 @@ print_usage() {
   echo "  -c, --current             Select the currently active window as target."
   echo "  -n, --name WINDOW_NAME    Specify and try to match a window name."
   echo "  -w, --window WINDOW_ID    Specify the window id of the target window."
-}
+} #}
 
-case "$0" in
-  *compton-trans*)  echo "Warning: compton has been renamed, please use picom-trans instead" >& 2;;
-esac
 
-# "command" is a shell built-in, faster than "which"
-if test -z "$(command -v xprop)" -o -z "$(command -v xwininfo)"; then
-  echo 'The command xwininfo or xprop is not available. They might reside in a package named xwininfo, xprop, x11-utils, xorg-xprop, or xorg-xwininfo.' >& 2
-  exit 1
-fi
+parse_args()
+{ #{
 
-if test $# -eq 0; then
-  print_usage >& 2
-  exit 1
-fi
+  # Iterate over every option and convert
+  # longopts (GNU) into POSIX-compatible opts
+  i=1       # because we start from "$1", not from "$0"
+  while [ $i -le $# ]
+  do
+    # Catch invalid options
+    case "$1" in
+      (--opacity=|--name=|--window=)
+        echo "Option $1 needs a value." >&2
+        exit 1
```

Comment:
Add a more descriptive error message if long options have a missing value. Otherwise `picom-trans --name` complains about `-n` not having a value.
```suggestion
      (--opacity=|--name=|--window=)
        echo "Option ${1%%=} needs a value." >&2
        exit 1;;
      (--opacity|--name|--window)
        [ $i -eq $# ] && echo "Option $1" needs a value. >&2 && exit 1
```

Reply:
Implemented. Thanks!